src/main/java/io/github/balconyseats/vertx/helper/kafka/KafkaHelper.java

```diff
@@ -0,0 +1,170 @@
+
+        consumerConfig.putAll(commonKafkaConfig);
+
+        return KafkaConsumer.create(vertx, consumerConfig);
+    }
+
+    /**
+     * Creates {@link KafkaProducer} from config object
+     * @param vertx vertx instance
+     * @param producerName consumer name in config
+     * @param config configuration
+     * @param <K> key type
+     * @param <V> value type
+     * @return {@link KafkaConsumer} instance
+     */
+    public static <K, V> KafkaProducer<K, V> createProducer(Vertx vertx, String producerName, JsonObject config) {
+
+        JsonObject kafkaConfig = kafkaConfig(config);
+
+        Map<String, String> commonKafkaConfig = commonKafkaConfig(kafkaConfig);
+        Map<String, String> producerConfig = producerConfig(producerName, kafkaConfig);
+
+        producerConfig.putAll(commonKafkaConfig);
+
+        return KafkaProducer.create(vertx, producerConfig);
+    }
+
+    /**
+     * Retrieves consumer topic from configuration
+     * @param consumerName consumer name
+     * @param config configuration
+     * @return topic name
+     */
+    public static String consumerTopic(String consumerName, JsonObject config) {
+        String topicPath = String.format("/kafka/consumers/%s/topic", consumerName);
+        return Optional.ofNullable(ConfigUtil.getString(topicPath, config))
+            .orElseThrow(() -> new IllegalConfigurationException(String.format("Kafka configuration for topic '%s' not exists.", topicPath)));
+    }
+
+    /**
+     * Retrieves producer topic from configuration
+     * @param producerName producer name
+     * @param config configuration
+     * @return topic name
+     */
+    public static String producerTopic(String producerName, JsonObject config) {
+        String topicPath = String.format("/kafka/producers/%s/topic", producerName);
+        return Optional.ofNullable(ConfigUtil.getString(topicPath, config))
+            .orElseThrow(() -> new IllegalConfigurationException(String.format("Kafka configuration for topic '%s' not exists.", topicPath)));
+    }
+
+    private static Map<String, String> consumerConfig(String consumerName, JsonObject kafkaConfig) {
+        return extractConfig("consumers", consumerName, kafkaConfig);
+    }
+
+    private static Map<String, String> producerConfig(String consumerName, JsonObject kafkaConfig) {
+        return extractConfig("producers", consumerName, kafkaConfig);
+    }
+
+    private static Map<String, String> extractConfig(String path, String name, JsonObject kafkaConfig) {
+        JsonObject consumersConfig = Optional.ofNullable(kafkaConfig)
+            .map(kc -> kc.getJsonObject(path))
+            .orElseThrow(() -> new IllegalConfigurationException(String.format("Kafka configuration not exists 'kafka.%s'.", path)));
+
+        JsonObject commonConfig = Optional.ofNullable(consumersConfig.getJsonObject("commonConfig")).orElse(new JsonObject());
+
+        JsonObject consumerConfig = Optional.ofNullable(consumersConfig.getJsonObject(name))
```

Comment:
I would use different name for `consumerConfig` variable, since `extractConfig` method is helper method and config is not `consumer` specific.